platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py

```diff
@@ -1061,6 +1064,8 @@ def get_temperature(self):
             0.0 if module temperature is not supported or module is under initialization
             other float value if module temperature is available
         """
+        if not self.get_presence():
```

Comment:
Maybe we could move the check to thermal.py class ModuleThermal. Usually, we prefer caller to check get_presence.

Reply:
Done